client-oauth2.js

```diff
@@ -703,6 +711,59 @@
   };
 
   /**
+   * Support JSON Web Token (JWT) Bearer Token OAuth 2.0 grant.
+   *
+   * Reference: https://tools.ietf.org/html/draft-ietf-oauth-jwt-bearer-12#section-2.1
+   *
+   * @param {ClientOAuth2} client
+   */
+  function JwtBearerFlow (client) {
+    this.client = client;
+  }
+
+  /**
+   * Request an access token using a JWT token.
+   *
+   * @param  {string} token A JWT token.
+   * @param  {Object}  [options]
+   * @return {Promise}
+   */
+  JwtBearerFlow.prototype.getToken = function (token, options) {
+    var self = this;
+
+    options = assign({}, this.client.options, options);
+
+    expects(options, [
+      'accessTokenUri'
+    ]);
+
+    var headers = {
+      'Accept':        'application/json, application/x-www-form-urlencoded',
+      'Content-Type':  'application/x-www-form-urlencoded'
+    };
+
+    // Authentication of the client is optional, as described in Section 3.2.1 of OAuth 2.0 [RFC6749]
+    if (options.clientId) {
+      headers['Authorization'] = 'Basic ' + btoa(options.clientId + ':' + options.clientSecret || '');
```

Comment:
Yes, I meant the wrap the client secret like `(options.clientSecret || '')`, because a client secret is not required for a public client. If I remember correctly, a empty password is acceptable in a HTTP basic authentication header. So this would allow public client ot use the library.